test.js

```diff
@@ -893,12 +893,75 @@ describe('hash', () => {
 describe('Create Error tests for Satellite', () => {
   test('should be an instance of type Error', () => {
     expect(createError(404, 'testing') instanceof Error).toBe(true);
+
+    const testError = createError(404, 'Test for Errors');
+    const nestedError = createError(testError);
+    expect(nestedError instanceof Error).toBe(true);
   });
 
   test("should have it's value, and message accessible through it's members", () => {
     const testError = createError(404, 'Satellite Test for Errors');
     expect(testError.status).toBe(404);
     expect(testError.message).toBe('Satellite Test for Errors');
+
+    const nestedError = createError(503, testError);
+    // error status of an Error instance will not be overwritten
+    expect(nestedError.status).toBe(404);
+    expect(nestedError.message).toBe('Satellite Test for Errors');
+  });
+
+  test('should create an Error from an object', () => {
+    const testObj = { key1: 'testing object', key2: 'for createError' };
+    const testError = createError(404, testObj);
+    expect(testError instanceof Error).toBe(true);
+    expect(testError.key1).toBe('testing object');
+    expect(testError.key2).toBe('for createError');
+  });
+
+  test('should fail when directly creating Error from ElasticSearch error object', () => {
+    // { errors } is imported from ElasticSearch
+    const elasticError = new errors.ResponseError({
+      body: { error: 'testing ElasticSearch Error' },
+      statusCode: 404,
+    });
+
+    try {
+      createError(503, elasticError);
```

Comment:
were bringing back a 503 because elastic returns 404 not available correct?

Reply:
I'm just replicating what's being done in the `query.js` that's causing the error
```javascript
// query.js
router.get('/', validateQuery, async (req, res, next) => {
  try {
    const { text, filter, page, perPage } = req.query;
    res.send(await search(text, filter, page, perPage));
  } catch (error) {
    console.log({ error }, 'query error');
    next(createError(503, error));   // <-- crash is here
  }
});
```
If you look below in the indirect test, there are different ways we can send in the "correct" status code. 
The thing is though, ES has different status code names for their errors than the regular http errors. Even if we send the ES status code in for `createError`, it'll just name it depending on the http error and not the ES error.